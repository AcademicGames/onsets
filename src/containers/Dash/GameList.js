import React from "react";
import PropTypes from "prop-types";
import firebase from "firebase";
import styled from "styled-components";
import { Link } from "react-router-dom";
import Panel from "muicss/lib/react/panel";
import Container from "muicss/lib/react/container";
import Button from "muicss/lib/react/button";

export default class Dash extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      games: {},
      ids: []
    };
  }

  componentWillMount() {
    this.fetchActiveGames();
  }

  render() {
    let arch = 0;
    return (
      <Container>
        <RowContainer>
          <TitlesWrapper>
            <div>Game ID#</div>
            <div>Players In Game</div>
            <div>Join</div>
          </TitlesWrapper>
          {this.state.ids.map((game, i) => {
            if (this.state.games[game].archived) {
              arch++;
              return null;
            } else {
              let playerCount = 0;
              let players = this.state.games[game].players;
              if (players) {
                //TODO: Is this an array or object?
                playerCount = Object.keys(players).length;
              }
              return (
                <RowPrimitive
                  key={"game-row-" + i}
                  className={(i - arch) % 2 == 0 ? "even" : "odd"}
                >
                  <LinkWrapper>
                    <Link to={`/game/${game}`}>
                      {game}
                    </Link>
                  </LinkWrapper>
                  <CountPrimitive>
                    {this.state.games[game].players ? playerCount : 0}
                  </CountPrimitive>
                  <Button
                    disabled={playerCount === 3}
                    variant="raised"
                    onClick={this.handleJoin.bind(this, game)}
                    style={{ margin: "0", position: "inherit" }}
                    color="primary"
                  >
                    Join
                  </Button>
                </RowPrimitive>
              );
            }
          })}
        </RowContainer>
      </Container>
    );
  }

  handleJoin = id => {
    let uid = firebase.auth().currentUser.uid;
    let displayName = firebase.auth().currentUser.displayName;
    let players = this.state.games[id].players;
    let i;
    //XXX: should players be added to a random position?
    if (players) {
      if (!players[0]) {
        i = 0;
      } else if (!players[1] && players[0] != uid) {
        i = 1;
      } else if (!players[2] && players[1] != uid && players[2] != uid) {
        i = 2;
      } else {
        return;
      }
      firebase.database().ref(`games/${id}/players/${i}`).transaction(data => {
        return { uid: uid, name: displayName };
      });
      if (i === 2) {
        firebase.database().ref(`games/${id}/stage`).transaction(data => {
          return "universe";
        });
      }
      this.props.history.push(`/game/${id}`);
    } else {
      firebase.database().ref(`games/${id}/players`).transaction(data => {
        return { 0: { uid: uid, name: displayName } };
      });
      this.props.history.push(`/game/${id}`);
    }
  };

  fetchActiveGames = () => {
    //TODO: FUTURE: Handle pagination
    firebase.database().ref(`games`).on(
      "value",
      snapshot => {
        let allGames = snapshot.val();
        this.setState({
          games: allGames,
          ids: Object.keys(allGames)
        });
      },
      err => {
        console.log(err);
      }
    );
  };
}

const RowContainer = styled.div`
  border-radius: 20px;
  box-shadow: 3px 5px 30px -4px rgba(0, 0, 0, 0.75);
  margin: 0 auto;
  margin-top: 80px;
  background-color: #f2f2f2;
  padding: 5px;
`;
export const RowPrimitive = styled.div`
  margin: 0 auto;
  margin-bottom: 10px;
  margin-top: 10px;
  padding: 1vh;
  width: 98%;
  height: 50px;
  border-radius: 5px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  transition: all 200ms ease-in-out;
  &:hover {
    box-shadow: 0 0 20px 5px rgba(0, 0, 0, 0.15);
  }
  &.even {
    background-color: rgba(247, 238, 127, 0.5);
  }
  &.odd {
    background-color: rgba(200, 70, 80, 0.5);
  }
`;
const TitlesWrapper = styled.div`
  width: 85%;
  margin: 0 auto;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;
const LinkWrapper = styled.div`
  font-family: monospace;
  line-height: 30px;
  font-size: 15px;
  line-height: 30px;
  display: inline-block;
`;
export const CountPrimitive = styled.div`
  width: 40px;
  font-size: 20px;
  height: 30px;
  margin-left: 0 auto;
  display: inline-block;
`;
