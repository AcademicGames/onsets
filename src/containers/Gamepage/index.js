// @flow

/**
 * Gamepage.
 * Static version of game, meant for players to use while in person
 */
import React from "react";
import type { Node } from "react";
import styled from "styled-components";

import Board from "components/Board";
import Universe from "components/Universe";
import Resources from "components/Cube/Resources";
import NavBar from "components/NavBar";
import ScratchPad from "components/ScratchPad";
import Variations from "components/Variations";
import UniverseSetup from "components/Universe/UniverseSetup";
import GoalArea from "components/GoalArea";
import Button from "muicss/lib/react/button";
import UniverseGrid from "components/Universe/UniverseGrid";
import AlertBar from "components/AlertBar";
import { rollCubes, shuffleCards } from "shared/game_setup_service";

import type { RouterHistory, Location, Match } from "react-router-dom";
type Props = { match: Match, location: Location, history: RouterHistory };
type State = GameStateOffline;

export default class Gamepage extends React.Component<Props, State> {
  state = {
    alert: undefined,
    cards: undefined,
    challenge: undefined,
    cubes: undefined,
    forbidden: [],
    goal: undefined,
    movingCube: undefined,
    permitted: [],
    required: [],
    stage: "new",
    stall: { began: 0 },
    turn: 0,
    variations: []
  };

  componentWillMount() {
    this.setState({ cubes: rollCubes(), stage: "universe" });
  }

  render() {
    return (
      <GameWrapper>
        <NavBar
          mode="game"
          history={this.props.history}
          stall={this.state.stall}
          resetStall={this.resetStall}
          setStall={this.setStall}
          turn={this.state.turn}
        />
        <AlertBar alert={this.state.alert} />
        <BoardPrimitive className={this.state.alert ? "alert" : ""}>
          <ResourcesWrapper>
            {Resources({
              isDisabled: this.state.challenge ? true : false,
              setMovingCube: this.setMovingCube,
              movingCube: this.state.movingCube,
              turn: this.state.turn,
              cubes: this.state.cubes
            })}
            {this.renderGoalArea()}
          </ResourcesWrapper>
          <ScratchPad mode="game" challenge={this.state.challenge} />
          {this.renderBoard()}
          {this.renderVariations()}
        </BoardPrimitive>
      </GameWrapper>
    );
  }

  // ----- Render Control ------

  renderGoalArea = (): Node => {
    if (["goal", "ingame", "challenge", "reveal"].includes(this.state.stage)) {
      return (
        <GoalArea
          cubes={this.state.cubes}
          goal={this.state.goal}
          isDisabled={this.state.stage === "goal"}
          isOnline={false}
          markUsed={this.markUsed}
          movingCube={this.state.movingCube}
          setGoal={this.setGoal}
          setMovingCube={this.setMovingCube}
        />
      );
    } else {
      return null;
    }
  };

  renderBoard = (): Node => {
    if (["ingame", "challenge", "reveal"].includes(this.state.stage)) {
      return Board({
        game: this.state,
        movingCube: this.state.movingCube,
        moveCubeTo: this.moveCubeTo
      });
    }
  };

  renderVariations = (): Node => {
    if (this.state.stage === "universe") {
      return <UniverseSetup setUniverse={this.setUniverse} />;
    } else if (["goal", "ingame", "challenge", "reveal"].includes(this.state.stage)) {
      return (
        <div>
          {Universe({ cards: this.state.cards })}
          <UniverseGrid cards={this.state.cards} />
          <Variations setVariation={this.setVariation} variations={this.state.variations} />
        </div>
      );
    } else if (this.state.stage === "variations") {
      return (
        <div>
          <Variations setVariation={this.setVariation} variations={this.state.variations} />
          {Universe({ cards: this.state.cards })}
          <UniverseGrid cards={this.state.cards} />
        </div>
      );
    }
  };

  // --------- Methods -------------

  markUsed = (cube: Cube): void => {
    if (!this.state.cubes) throw new Error("Internal Rep Error");
    let numbers = this.state.cubes.numbers;
    for (var i = 0; i < numbers.length; i++) {
      if (numbers[i].index === cube.index) {
        numbers[i].wasUsed = true;
      }
    }
    let cubes = this.state.cubes;
    cubes.numbers = numbers;
    this.setState({ cubes });
  };

  setUniverse = (size: number) => {
    this.setState({ universe: size });
    this.setState({ cards: shuffleCards(size) });
    this.setState({ stage: "variations" });
  };

  resetStall = () => {
    this.setState({ stall: { began: 0 } });
  };

  setStall = () => {
    if (this.state.stall.began === 0) {
      let stall = {
        began: new Date().getTime()
      };
      this.setState({ stall: stall });
    }
  };

  setVariation = (variation: string): void => {
    let variations = this.state.variations;
    if (!this.state.variations[0]) {
      variations[0] = variation;
    } else if (this.state.variations[1] == null) {
      variations[1] = variation;
    } else if (this.state.variations[2] == null) {
      variations[2] = variation;
      this.setState({ stage: "goal" });
    } else {
      return;
    }
    this.setState({ variations: variations });
  };

  endTurn = () => {
    this.setState({ turn: this.state.turn + 1 });
  };

  setGoal = (goal: Goal, value: number) => {
    this.setState({ goal: goal });
    this.setState({ stage: "ingame" });
    this.setState({ goal_value: value });
    this.endTurn();
  };

  setMovingCube = (cube?: Cube) => {
    if (!cube) this.setState({ movingCube: undefined });
    else this.setState({ movingCube: cube });
  };

  moveCubeTo = (category: BoardArea): void => {
    const movingCube = this.state.movingCube;
    if (movingCube) {
      let update = {};
      update[category] = this.state[category].concat([movingCube]);
      this.setState(update);
      this.setMovingCube();
      this.removeFromResources(movingCube);
    }
  };

  removeFromResources = (cube: Cube): void => {
    let cubes = this.state.cubes;
    if (!cubes || !this.state.cubes) return;
    let newResources = this.state.cubes[cube.type].filter(resource => {
      return resource.index !== cube.index;
    });
    cubes[cube.type] = (newResources: any);
    this.setState({ cubes: cubes });
    this.endTurn();
  };
}

const ResourcesWrapper = styled.div`
  width: 750px;
  height: 180px;
  margin-top: 70px;
`;
const GameWrapper = styled.div`
  padding-bottom: 80px;
`;
const BoardPrimitive = styled.div`
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  width: 800px;
  padding: 0 16px;
`;
