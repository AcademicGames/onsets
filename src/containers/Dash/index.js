/**
 * Dash
 * This component acts as a user's dashboard once they've logged in. Here they can see available
 * games, create new games, and access/modify user information.
 */
import React from "react";
import PropTypes from "prop-types";
import firebase from "firebase";
import styled from "styled-components";
import LogIn from "containers/LogIn";
import GameList from "./GameList";
import Loading from "components/Loading";
import NavBar from "components/NavBar";

import Button from "muicss/lib/react/button";

export default class Dash extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      auth: null
    };
  }

  componentWillMount() {
    this.attachAuthListener();
  }

  render() {
    if (this.state.auth) {
      return (
        <div>
          <NavBar mode="dash" history={this.props.history} />
          <GameList history={this.props.history} />
          <ButtonWrapper>
            <Button
              variant="fab"
              color="primary"
              onClick={this.handleAddGame}
              style={newGameStyle}
            >
              {" "}+{" "}
            </Button>
          </ButtonWrapper>
        </div>
      );
    } else if (this.state.auth === null) {
      return <Loading />;
    } else {
      return <LogIn />;
    }
  }

  handleAddGame = () => {
    let newGame = firebase.database().ref("games").push();
    newGame.set({ stage: "new" });
  };

  attachAuthListener = () => {
    firebase.auth().onAuthStateChanged(user => {
      if (user) {
        this.setState({ auth: true });
      } else {
        this.setState({ auth: false });
      }
    });
  };
}

// --- Styles ---

//TODO: make + spin
const newGameStyle = {
  fontSize: "30px",
  margin: "0"
};

const AddGame = styled.div`
  position: absolute;
  right: 50px;
  bottom: 50px;
  width: 50px;
  height: 50px;
  border-radius: 25px;
  color: white;
  background-color: blue;
  text-align: center;
  font-size: 30px;
  vertical-align: middle;
`;

const ButtonWrapper = styled.div`
  position: absolute;
  bottom: 5%;
  right: 5%;
  width: 55px;
  height: 55px;
  border-radius: 50%;
  transition: transform 150ms ease-in-out, box-shadow 200ms 0ms ease-in;
  &:hover {
    transform: rotate(360deg) scale(1.05);
    box-shadow: 0px 0px 25px 2px rgba(0, 0, 0, 0.3);
  }
`;
