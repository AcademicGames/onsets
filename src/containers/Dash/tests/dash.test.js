import React from "react";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import Gamepage from "../index.js";
import { MemoryRouter } from "react-router-dom";
import { DragDropContext } from "react-dnd";
import { RowPrimitive, CountPrimitive } from "../GameList";
import TestBackend from "react-dnd-test-backend";
import "jest-styled-components";
import toJson from "enzyme-to-json";
import Dash from "../";

import fb from "firebase";
jest.mock("firebase");

function wrapInTestContext(DecoratedComponent) {
  return DragDropContext(TestBackend)(
    class TestContextContainer extends React.Component {
      render() {
        return <DecoratedComponent {...this.props} />;
      }
    }
  );
}
const injectMock = data => {
  fb.setMock(data);
  let WrappedGamepage = wrapInTestContext(Dash);
  return mount(
    <MemoryRouter>
      <WrappedGamepage match={{ params: { gameId: "id1" } }} />
    </MemoryRouter>
  );
};
const games = {
  id1: {},
  id2: { players: [{ name: "steph", uid: "uid1" }] }
};
describe("The Dash", () => {
  it("shows a list of active games", () => {
    let wrapper = injectMock({ game: games, auth: "uid1" });
    expect(wrapper.find(RowPrimitive).length).toEqual(2);
  });
  it("shows the number of players in a game", () => {
    let wrapper = injectMock({ game: games, auth: "uid1" });
    const count = wrapper.find(RowPrimitive).at(1).find(CountPrimitive);
    expect(count.text()).toEqual("1");
    // must find the component under the router for snapshot
    expect(toJson(wrapper.find(Dash))).toMatchSnapshot();
  });
});
