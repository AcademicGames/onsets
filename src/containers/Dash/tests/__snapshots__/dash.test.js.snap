// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Dash shows the number of players in a game 1`] = `
.c0 {
  background-color: #7f0000;
  box-shadow: 0px -2px 22px 10px rgba(0,0,0,0.75);
  color: #f7ee7f;
  height: 40px;
  margin-bottom: 20px;
  position: fixed;
  top: 0px;
  width: 100%;
  z-index: 200;
}

.c1 {
  position: absolute;
  width: 100%;
}

.c2 {
  color: #ffa726;
  margin: 0 auto;
  width: 150px;
  height: 40px;
  text-align: center;
  font-weight: 600;
  font-size: 30px;
  -webkit-transition: all 400ms ease-in-out;
  transition: all 400ms ease-in-out;
  cursor: pointer;
  text-shadow: 1px 1px 0 black;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2:hover {
  color: yellow;
}

.c3 {
  position: absolute;
  cursor: pointer;
  top: 10px;
  right: 20px;
}

.c4 {
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
}

.c4:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c5 {
  opacity: 0;
  cursor: pointer;
  color: #e5e7e9;
  border-radius: 10px;
  position: absolute;
  top: -130px;
  width: 210px;
  background: #2e4053;
  z-index: 98;
  right: -20px;
  height: 100px;
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
  box-shadow: 0px 5px 30px -4px rgba(0,0,0,0.75);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c5.open {
  -webkit-transform: translate(0px,180px);
  -ms-transform: translate(0px,180px);
  transform: translate(0px,180px);
  opacity: 1;
}

.c5:focus {
  outline: none;
}

.c6 {
  position: absolute;
  top: -8px;
  right: 35px;
  width: 0;
  height: 0;
  border-bottom: 8px solid #2e4053;
  border-right: 8px solid transparent;
  border-left: 8px solid transparent;
}

.c7 {
  width: 90%;
  text-align: center;
  height: 15px;
  margin: 0 auto;
  font-size: 16px;
}

.c8 {
  border-radius: 20px;
  box-shadow: 3px 5px 30px -4px rgba(0,0,0,0.75);
  margin: 0 auto;
  margin-top: 80px;
  background-color: #f2f2f2;
  padding: 5px;
}

.c9 {
  width: 85%;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
}

.c11 {
  margin: 0 auto;
  margin-bottom: 10px;
  margin-top: 10px;
  padding: 1vh;
  width: 98%;
  height: 50px;
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
}

.c11:hover {
  box-shadow: 0 0 20px 5px rgba(0,0,0,0.15);
}

.c11.c10 {
  background-color: rgba(247,238,127,0.5);
}

.c11.c14 {
  background-color: rgba(200,70,80,0.5);
}

.c12 {
  font-family: monospace;
  line-height: 30px;
  font-size: 15px;
  line-height: 30px;
  display: inline-block;
}

.c13 {
  width: 40px;
  font-size: 20px;
  height: 30px;
  margin-left: 0 auto;
  display: inline-block;
}

.c15 {
  position: absolute;
  bottom: 5%;
  right: 5%;
  width: 55px;
  height: 55px;
  border-radius: 50%;
  -webkit-transition: -webkit-transform 150ms ease-in-out,box-shadow 200ms 0ms ease-in;
  -webkit-transition: transform 150ms ease-in-out,box-shadow 200ms 0ms ease-in;
  transition: transform 150ms ease-in-out,box-shadow 200ms 0ms ease-in;
}

.c15:hover {
  -webkit-transform: rotate(360deg) scale(1.05);
  -ms-transform: rotate(360deg) scale(1.05);
  transform: rotate(360deg) scale(1.05);
  box-shadow: 0px 0px 25px 2px rgba(0,0,0,0.3);
}

<Dash
  match={
    Object {
      "params": Object {
        "gameId": "id1",
      },
    }
  }
>
  <div>
    <NavBar
      mode="dash"
    >
      <styled.div>
        <div
          className="c0"
        >
          <styled.div>
            <div
              className="c1"
            >
              <styled.div
                onClick={[Function]}
              >
                <div
                  className="c2"
                  onClick={[Function]}
                >
                  On Sets
                </div>
              </styled.div>
            </div>
          </styled.div>
          <styled.div
            onClick={[Function]}
          >
            <div
              className="c3"
              onClick={[Function]}
            >
              <styled.div>
                <div
                  className="c4"
                >
                  <svg
                    height="14px"
                    viewBox="0 0 24 14"
                    width="24px"
                  >
                    <rect
                      height="2"
                      rx="1"
                      width="17"
                      x="7"
                      y="0"
                    />
                    <rect
                      height="2"
                      rx="1"
                      width="24"
                      x="0"
                      y="6"
                    />
                    <rect
                      height="2"
                      rx="1"
                      width="24"
                      x="0"
                      y="12"
                    />
                  </svg>
                </div>
              </styled.div>
              <DropDown
                isOpen={false}
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <styled.div
                  className=""
                  id="menu"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  tabIndex="0"
                >
                  <div
                    className="c5"
                    id="menu"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    tabIndex="0"
                  >
                    <styled.div>
                      <div
                        className="c6"
                      />
                    </styled.div>
                    <styled.div
                      onClick={[Function]}
                    >
                      <div
                        className="c7"
                        onClick={[Function]}
                      >
                        Log Out
                      </div>
                    </styled.div>
                    <styled.div>
                      <div
                        className="c7"
                      >
                         Edit Profile 
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </DropDown>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </NavBar>
    <Dash>
      <Container
        className=""
        fluid={false}
      >
        <div
          className="mui-container "
        >
          <styled.div>
            <div
              className="c8"
            >
              <styled.div>
                <div
                  className="c9"
                >
                  <div>
                    Game ID#
                  </div>
                  <div>
                    Players In Game
                  </div>
                  <div>
                    Join
                  </div>
                </div>
              </styled.div>
              <styled.div
                className="c10"
              >
                <div
                  className="c10 c11"
                >
                  <styled.div>
                    <div
                      className="c12"
                    >
                      <Link
                        replace={false}
                        to="/game/id1"
                      >
                        <a
                          href="/game/id1"
                          onClick={[Function]}
                        >
                          id1
                        </a>
                      </Link>
                    </div>
                  </styled.div>
                  <styled.div>
                    <div
                      className="c13"
                    >
                      0
                    </div>
                  </styled.div>
                  <Button
                    className=""
                    color="primary"
                    disabled={false}
                    onClick={[Function]}
                    size="default"
                    style={
                      Object {
                        "margin": "0",
                        "position": "inherit",
                      }
                    }
                    variant="raised"
                  >
                    <button
                      className="mui-btn mui-btn--primary mui-btn--raised "
                      disabled={false}
                      onClick={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      style={
                        Object {
                          "margin": "0",
                          "position": "inherit",
                        }
                      }
                    >
                      Join
                      <span
                        className="mui-btn__ripple-container"
                      >
                        <span
                          className="mui-ripple"
                          style={Object {}}
                        />
                      </span>
                    </button>
                  </Button>
                </div>
              </styled.div>
              <styled.div
                className="c14"
              >
                <div
                  className="c14 c11"
                >
                  <styled.div>
                    <div
                      className="c12"
                    >
                      <Link
                        replace={false}
                        to="/game/id2"
                      >
                        <a
                          href="/game/id2"
                          onClick={[Function]}
                        >
                          id2
                        </a>
                      </Link>
                    </div>
                  </styled.div>
                  <styled.div>
                    <div
                      className="c13"
                    >
                      1
                    </div>
                  </styled.div>
                  <Button
                    className=""
                    color="primary"
                    disabled={false}
                    onClick={[Function]}
                    size="default"
                    style={
                      Object {
                        "margin": "0",
                        "position": "inherit",
                      }
                    }
                    variant="raised"
                  >
                    <button
                      className="mui-btn mui-btn--primary mui-btn--raised "
                      disabled={false}
                      onClick={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      style={
                        Object {
                          "margin": "0",
                          "position": "inherit",
                        }
                      }
                    >
                      Join
                      <span
                        className="mui-btn__ripple-container"
                      >
                        <span
                          className="mui-ripple"
                          style={Object {}}
                        />
                      </span>
                    </button>
                  </Button>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </div>
      </Container>
    </Dash>
    <styled.div>
      <div
        className="c15"
      >
        <Button
          className=""
          color="primary"
          onClick={[Function]}
          size="default"
          style={
            Object {
              "fontSize": "30px",
              "margin": "0",
            }
          }
          variant="fab"
        >
          <button
            className="mui-btn mui-btn--primary mui-btn--fab "
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "fontSize": "30px",
                "margin": "0",
              }
            }
          >
             
            +
             
            <span
              className="mui-btn__ripple-container"
            >
              <span
                className="mui-ripple"
                style={Object {}}
              />
            </span>
          </button>
        </Button>
      </div>
    </styled.div>
  </div>
</Dash>
`;
